homeassistant/components/adguard/__init__.py

```diff
@@ -41,6 +43,15 @@
 )
 
 PLATFORMS = [Platform.SENSOR, Platform.SWITCH]
+ADGUARD_HASS_KEY = HassEntryKey["AdGuardData"]()
```

Comment:
The downside of this approach is that the key itself is now an object, and so if we ever see garbage data in `hass.data`, we wouldn't know which object generated it. I can't remember if that has been an issue in the past. It will probably break a bunch of custom components doing hacky things on other integrations `hass.data` that they probably shouldn't do anyway 🙃

Reply:
It would be fairly easy to add debugging information to the key object. Either a string or even the module name would work.

_The question for me would be if this idea is something we want to pursue further. I.e. should I invest some more time in fixing the overloads?_

Reply:
I think with the last push this is ok since will be a subclassed `str` at runtime.  Not my review comment though so I didn't want to resolve it.